php/media/class-gallery.php

```diff
@@ -398,6 +391,47 @@ protected function register_settings() {
 		$media_settings->add_setting( $this->settings );
 	}
 
+	/**
+	 * Inits the cloudinary gallery using block attributes.
+	 *
+	 * @param string $content The post content.
+	 * @param array  $block   Block data.
+	 *
+	 * @return string
+	 */
+	public function prepare_block_render( $content, $block ) {
+		if ( 'cloudinary/gallery' !== $block['blockName'] ) {
+			return $content;
+		}
+
+		$attributes = Utils::expand_dot_notation( $block['attrs'], '_' );
+		$attributes = array_merge( self::$default_config, $attributes );
+
+		// Gallery without images. Don't render.
+		if ( empty( $attributes['selectedImages'] ) ) {
+			return $content;
+		}
+
+		$attributes['mediaAssets'] = $attributes['selectedImages'];
+		$attributes['cloudName']   = $this->media->plugin->components['connect']->get_cloud_name();
+		unset( $attributes['selectedImages'], $attributes['customSettings'] );
+
+		ob_start();
+		?>
+		<script>
+			window.addEventListener( 'load', function () {
+				if ( cloudinary?.galleryWidget ) {
```

Comment:
@dugajean 
Given that this is not going through Babel, and that [optional changing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is only supported in recent versions of browsers, I think we should be a bit more explicit here.

Reply:
🤦🏻